shared/src/main/scala/FuncHelpers.scala

```diff
@@ -10,24 +10,51 @@ object FuncHelpers:
       fn.arity match
         case 1 =>
           val a = ctx.pop()
-          Monad.vectoriseNoFill { a =>
+          VList(ListHelpers.makeIterable(a).map { a =>
             Interpreter.executeFn(fn, args = Some(List(a)))
-          }(a)
+          }*)
         case 2 =>
           val b, a = ctx.pop()
-          Dyad.vectoriseNoFill { (a, b) =>
-            Interpreter.executeFn(fn, args = Some(List(a, b)))
-          }(a, b)
+          VList(ListHelpers.makeIterable(a).zipWithIndex.map { (a, i) =>
+            Interpreter.executeFn(
+              fn,
+              args = Some(List(a, b))
```

Comment:
Perhaps it was a bad idea to take an `Option[List[VAny]]` for the arguments. We could use `List[VAny] | Null` instead to avoid the `Some()`. What do you think?

Reply:
Honestly, I'm just going off what copilot suggested. How would this look with `List[VAny]|Null`?

Reply:
There wouldn't be a `Some()`, we'd just pass `args = List(a, b)`

Reply:
That does sound good